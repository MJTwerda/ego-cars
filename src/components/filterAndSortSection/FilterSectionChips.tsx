import { vehicleFilterTabs } from '@/data/vehicleFilterTabs';
import { Chip, Typography } from '@mui/material';

export interface FilterSectionLabelI {
  label: string;
  isStrong?: boolean;
}

interface FilterSectionChipsProps {
  handleFilterModels: (tabValue: string) => void;
  filterLabel: FilterSectionLabelI;
}
const FilterSectionChips = ({ handleFilterModels, filterLabel }: FilterSectionChipsProps) => {

  return (
    <div className='flex w-2/4 sm:w-3/4'>
      <Typography variant='body1' className="flex self-center mr-4 sm:flex">
        {filterLabel.isStrong ? <strong>{filterLabel.label}</strong> : filterLabel.label}
      </Typography>

      {vehicleFilterTabs.map((tab) => (
        <div key={tab.value} className='flex self-center hidden lg:flex lg:justify-around lg:mr-4'>
          <Chip
            label={tab.label}
            onClick={() => handleFilterModels(tab.value)}
          />
        </div>
      ))}
    </div>
  )
};

export default FilterSectionChips;