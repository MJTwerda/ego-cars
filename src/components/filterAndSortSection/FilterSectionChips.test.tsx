import { render, screen, fireEvent, RenderResult, act, waitFor } from '@testing-library/react';
import FilterSectionChips from './FilterSectionChips';
import '@testing-library/jest-dom';
import { Chip } from '@mui/material';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import ExpandLessIcon from '@mui/icons-material/ExpandLess';
import { FilterSectionChipsProps } from './FilterSectionChips';

const mockLabel = 'Filtrar por';
const mockHandleFilterModels = jest.fn();

const mockProps: FilterSectionChipsProps = {
  handleFilterModels: mockHandleFilterModels,
  filterLabel: {
    label: mockLabel,
    isStrong: true
  }
}

describe('FilterSectionChips', () => {
  it('should render the filter section label correctly', async () => {
    let component: RenderResult;

    await act(() => {
      component =render(
        <FilterSectionChips 
          handleFilterModels={mockProps.handleFilterModels} 
          filterLabel={mockProps.filterLabel} 
        />
      );
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();
      const label = screen.getByText(mockLabel);
      expect(label).toBeInTheDocument();
      expect(label).toContainHTML('<strong>Filtrar por</strong>');
    })
  });

  it('should render the filter section label correctly', async () => {
    const newProps = {
      ...mockProps,
      filterLabel: {
        label: mockLabel,
        isStrong: false
      }
    }
    let component: RenderResult;

    await act(() => {
      component =render(
        <FilterSectionChips 
          handleFilterModels={newProps.handleFilterModels} 
          filterLabel={newProps.filterLabel} 
        />
      );
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();
      const label = screen.getByText(mockLabel);
      expect(label).toBeInTheDocument();
      expect(label).not.toContainHTML('<strong>Filtrar por</strong>');
    })
  });

  it('clicking on option, the handleFilterModels function should be called', async() => {
    let component: RenderResult;

    await act(() => {
      component =render(
        <FilterSectionChips 
          handleFilterModels={mockProps.handleFilterModels} 
          filterLabel={mockProps.filterLabel} 
        />
      );
    });

    await waitFor(() => {
      const optionAll = component.getByText('Todos');
      expect(optionAll).toBeInTheDocument();
      fireEvent.click(optionAll);
      expect(mockHandleFilterModels).toHaveBeenCalledTimes(1);
      expect(mockHandleFilterModels).toHaveBeenCalledWith('all');
    });
  });

  it('should handle clicking on the menu items in mobile view', async() => {
    let component: RenderResult;

    await act(() => {
      component =render(
        <FilterSectionChips 
          handleFilterModels={mockProps.handleFilterModels} 
          filterLabel={mockProps.filterLabel} 
        />
      );
    });

    await waitFor(() => {
      fireEvent.click(component.getByTestId('common_menu_icon_btn'));
  
      const menuItems = component.getAllByRole('menuitem');
      fireEvent.click(menuItems[0]); // Click on "Todos"
      expect(mockHandleFilterModels).toHaveBeenCalledWith('all');
  
      fireEvent.click(menuItems[1]); // Click on "Autos"
      expect(mockHandleFilterModels).toHaveBeenCalledWith('Sedan');
  
      fireEvent.click(menuItems[2]); // Click on "Pickups y Comerciales"
      expect(mockHandleFilterModels).toHaveBeenCalledWith('Pickups y Comerciales');
  
      fireEvent.click(menuItems[3]); // Click on "SUVs y Crossovers"
      expect(mockHandleFilterModels).toHaveBeenCalledWith('SUVs');
    });

  });
});
