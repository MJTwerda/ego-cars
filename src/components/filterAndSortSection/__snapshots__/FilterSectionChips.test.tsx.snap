// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSectionChips should render the filter section label correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-2/4 sm:w-3/4"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 flex self-center css-rizt0-MuiTypography-root"
        >
          <strong>
            Filtrar por
          </strong>
        </p>
        <div
          class="flex lg:hidden"
        >
          <div
            class="flex"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 hidden sm:flex self-center w-3/4 css-rizt0-MuiTypography-root"
            />
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
              data-testid="common_menu_icon_btn"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_all"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Todos
            </span>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_Sedan"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Autos
            </span>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_Pickups y Comerciales"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Pickups y Comerciales
            </span>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_SUVs"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              SUVs y Crossovers
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-2/4 sm:w-3/4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 flex self-center css-rizt0-MuiTypography-root"
      >
        <strong>
          Filtrar por
        </strong>
      </p>
      <div
        class="flex lg:hidden"
      >
        <div
          class="flex"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 hidden sm:flex self-center w-3/4 css-rizt0-MuiTypography-root"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="common_menu_icon_btn"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_all"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Todos
          </span>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_Sedan"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Autos
          </span>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_Pickups y Comerciales"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Pickups y Comerciales
          </span>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_SUVs"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            SUVs y Crossovers
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FilterSectionChips should render the filter section label correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-2/4 sm:w-3/4"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 flex self-center css-rizt0-MuiTypography-root"
        >
          Filtrar por
        </p>
        <div
          class="flex lg:hidden"
        >
          <div
            class="flex"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 hidden sm:flex self-center w-3/4 css-rizt0-MuiTypography-root"
            />
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
              data-testid="common_menu_icon_btn"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_all"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Todos
            </span>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_Sedan"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Autos
            </span>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_Pickups y Comerciales"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              Pickups y Comerciales
            </span>
          </div>
        </div>
        <div
          class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
            data-testid="filter_section_chips_SUVs"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              SUVs y Crossovers
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-2/4 sm:w-3/4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 flex self-center css-rizt0-MuiTypography-root"
      >
        Filtrar por
      </p>
      <div
        class="flex lg:hidden"
      >
        <div
          class="flex"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 hidden sm:flex self-center w-3/4 css-rizt0-MuiTypography-root"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="common_menu_icon_btn"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_all"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Todos
          </span>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_Sedan"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Autos
          </span>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_Pickups y Comerciales"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            Pickups y Comerciales
          </span>
        </div>
      </div>
      <div
        class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
          data-testid="filter_section_chips_SUVs"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
          >
            SUVs y Crossovers
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
